instrumentation/r2dbc-1.0/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/r2dbc/v1_0/AbstractR2dbcStatementTest.java

```diff
@@ -0,0 +1,261 @@
+              "MYSQL_USER", USER_DB,
+              "MYSQL_PASSWORD", PW_DB,
+              "MYSQL_DATABASE", DB);
+
+  private static final DbSystemProps MYSQL =
+      new DbSystemProps("mysql", "mysql:8.0.32", 3306)
+          .envVariables(
+              "MYSQL_ROOT_PASSWORD", PW_DB,
+              "MYSQL_USER", USER_DB,
+              "MYSQL_PASSWORD", PW_DB,
+              "MYSQL_DATABASE", DB);
+
+  private static final Map<String, DbSystemProps> SYSTEMS = new HashMap<>();
+
+  static {
+    SYSTEMS.put(POSTGRESQL.system, POSTGRESQL);
+    SYSTEMS.put(MYSQL.system, MYSQL);
+    SYSTEMS.put(MARIADB.system, MARIADB);
+  }
+
+  private static Integer port;
+  private static GenericContainer<?> container;
+
+  abstract ConnectionFactory createProxyConnectionFactory(
+      ConnectionFactoryOptions connectionFactoryOptions);
+
+  void startContainer(DbSystemProps props) {
+    if (container != null && container.getDockerImageName().equals(props.image)) {
+      return;
+    }
+    if (container != null) {
+      container.stop();
+    }
+    if (props.image != null) {
+      container =
+          new GenericContainer<>(props.image)
+              .withEnv(props.envVariables)
+              .withExposedPorts(props.port)
+              .withLogConsumer(new Slf4jLogConsumer(logger))
+              .withStartupTimeout(Duration.ofMinutes(2));
+      container.start();
+      port = container.getMappedPort(props.port);
+    }
+  }
+
+  @ParameterizedTest(name = "{index}: {0}")
+  @MethodSource("provideParameters")
+  void testQueries(Parameter parameter) {
+    DbSystemProps props = SYSTEMS.get(parameter.system);
+    startContainer(props);
+    ConnectionFactory connectionFactory =
+        createProxyConnectionFactory(
+            ConnectionFactoryOptions.builder()
+                .option(DRIVER, props.system)
+                .option(HOST, "localhost")
+                .option(PORT, port)
+                .option(USER, USER_DB)
+                .option(PASSWORD, PW_DB)
+                .option(DATABASE, DB)
+                .build());
+
+    getTesting()
+        .runWithSpan(
+            "parent",
+            () -> {
+              Mono.from(connectionFactory.create())
+                  .flatMapMany(
+                      connection ->
+                          Mono.from(connection.createStatement(parameter.statement).execute())
+                              // Subscribe to the Statement.execute()
+                              .flatMapMany(result -> result.map((row, metadata) -> ""))
+                              .concatWith(Mono.from(connection.close()).cast(String.class)))
+                  .doFinally(e -> getTesting().runWithSpan("child", () -> {}))
+                  .blockLast();
+            });
+
+    getTesting()
+        .waitAndAssertTraces(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    span -> span.hasName("parent").hasKind(SpanKind.INTERNAL)),
+            trace ->
```

Comment:
You might want to add 
```kotlin
  testInstrumentation(project(":instrumentation:reactor:reactor-3.1:javaagent"))
```
to the gradle build file to stitch the traces together.